metron-interface/metron-alerts/src/app/service/elasticsearch-localstorage-impl.ts

```diff
@@ -0,0 +1,294 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Observable} from 'rxjs/Rx';
+import {Headers, RequestOptions} from '@angular/http';
+
+import {HttpUtil} from '../utils/httpUtil';
+import {DataSource} from './data-source';
+import {Alert} from '../model/alert';
+import {ColumnMetadata} from '../model/column-metadata';
+import {ElasticsearchUtils} from '../utils/elasticsearch-utils';
+import {
+  ALERTS_COLUMN_NAMES, ALERTS_TABLE_METADATA, ALERTS_RECENT_SEARCH,
+  ALERTS_SAVED_SEARCH, NUM_SAVED_SEARCH
+} from '../utils/constants';
+import {ColumnNames} from '../model/column-names';
+import {ColumnNamesService} from './column-names.service';
+import {TableMetadata} from '../model/table-metadata';
+import {SaveSearch} from '../model/save-search';
+import {AlertsSearchResponse} from '../model/alerts-search-response';
+import {SearchRequest} from '../model/search-request';
+
+export class ElasticSearchLocalstorageImpl extends DataSource {
+
+  private defaultColumnMetadata = [
+    new ColumnMetadata('_id', 'string'),
+    new ColumnMetadata('timestamp', 'date'),
+    new ColumnMetadata('source:type', 'string'),
+    new ColumnMetadata('ip_src_addr', 'ip'),
+    new ColumnMetadata('enrichments:geo:ip_dst_addr:country', 'string'),
+    new ColumnMetadata('ip_dst_addr', 'ip'),
+    new ColumnMetadata('host', 'string'),
+    new ColumnMetadata('alert_status', 'string')
+  ];
+
+  getAlerts(searchRequest: SearchRequest): Observable<AlertsSearchResponse> {
+    let url = '/search/*,-*' + ElasticsearchUtils.excludeIndexName + '/_search';
+    let request: any  = searchRequest;
+    request.query = { query_string: { query: searchRequest.query } };
+
+    return this.http.post(url, request, new RequestOptions({headers: new Headers(this.defaultHeaders)}))
+      .map(HttpUtil.extractData)
+      .map(ElasticsearchUtils.extractAlertsData)
+      .catch(HttpUtil.handleError)
+      .onErrorResumeNext();
+  }
+
+  getAlert(index: string, type: string, alertId: string): Observable<Alert> {
+    return this.http.get('/search/' + index + '/' + type + '/' + alertId, new RequestOptions({headers: new Headers(this.defaultHeaders)}))
+      .map(HttpUtil.extractData);
+  }
+
+  updateAlertState(request: any) {
+    return this.http.post('/search/_bulk', request, new RequestOptions({headers: new Headers(this.defaultHeaders)}))
+      .map(HttpUtil.extractData)
+      .catch(HttpUtil.handleError);
+  }
+
+  getDefaultAlertTableColumnNames(): Observable<ColumnMetadata[]> {
+    return Observable.create(observer => {
+      observer.next(JSON.parse(JSON.stringify(this.defaultColumnMetadata)));
+      observer.complete();
+    });
+  }
+
+  getAllFieldNames(): Observable<ColumnMetadata[]> {
+    let url = '_cluster/state';
```

Comment:
using cluster state here to enumerate indexes adds a lot of overhead, and brings back a lot of unnecessary meta data. Of course really this should ultimately be fronted with its own API from the REST service, but it might also be worth considering a simple '*' as url, which will just bring back index details without all the node and replica details.